plugins/gradle/gradle-plugin/src/main/java/org/wildfly/swarm/plugin/gradle/StartTask.java

```diff
@@ -0,0 +1,354 @@
+        try {
+
+            File tmp;
+            try {
+                tmp = Files.createTempFile("thorntail-process-file", null).toFile();
+                getLogger().info("Process file: " + tmp.toPath().toString());
+            } catch (IOException e) {
+                throw new GradleException("Error while creating process file");
+            }
+
+            process = executor.withDebug(extension.getDebugPort())
+                    .withProcessFile(tmp)
+                    .withProperties(extension.getProperties())
+                    .withProperties(ThorntailUtils.getPropertiesFromFile(extension.getPropertiesFile()))
+                    .withStdoutFile(extension.getStdoutFile() != null ? getProject().file(extension.getStdoutFile()).toPath() : null)
+                    .withStderrFile(extension.getStderrFile() != null ? getProject().file(extension.getStderrFile()).toPath() : null)
+                    .withEnvironment(extension.getEnvironment())
+                    .withEnvironment(ThorntailUtils.getPropertiesFromFile(extension.getEnvironmentFile()))
+                    .withWorkingDirectory(getProject().getProjectDir().toPath())
+                    .withProperty("remote.maven.repo",
+                            getProject().getRepositories().withType(MavenArtifactRepository.class).stream()
+                                    .map(mavenArtifactRepository -> mavenArtifactRepository.getUrl().toASCIIString())
+                                    .collect(Collectors.joining(",")))
+                    .execute();
+            registerCancellationListeners(process);
+
+            process.awaitReadiness(extension.getStartTimeout(), TimeUnit.SECONDS);
+
+            if (!process.isAlive()) {
+                throw new GradleException("Process failed to start");
+            }
+            if (process.getError() != null) {
+                throw new GradleException("Error starting process", process.getError());
+            }
+
+        } catch (IOException e) {
+            throw new GradleException("unable to execute", e);
+        } catch (InterruptedException e) {
+            throw new GradleException("Error waiting for deployment", e);
+        }
+
+        if (waitForProcess) {
+            try {
+                process.waitFor();
+            } catch (InterruptedException e) {
+                try {
+                    process.stop(10, TimeUnit.SECONDS);
+                } catch (InterruptedException ie) {
+                    // Do nothing
+                }
+            } finally {
+                process.destroyForcibly();
+            }
+        } else {
+            // save the process into a property so it can be stopped by another task later
+            getProject().getExtensions().getExtraProperties().set(PackagePlugin.THORNTAIL_PROCESS_PROPERTY, process);
+        }
+    }
+
+    private SwarmExecutor uberJarExecutor() {
+        getLogger().info("Creating uber-jar Thorntail executor.");
```

Comment:
With so many `info` logs, maybe we should consider prefixing the logs with something like `[THORNTAIL] `. That would make the logs easily searchable. Like in a CI system.